source-container-build/app/source_build.py

```diff
@@ -820,8 +1026,11 @@ def build(args) -> BuildResult:
 
     image_output_dir = build_source_image_in_local(args.bsi, work_dir, sib_dirs)
     if parent_sources_dir:
+        if build_result["dependencies_included"]:
+            deduplicate_sources(parent_sources_dir, image_output_dir)
```

Comment:
For my info, why do we de-duplicate only when dependencies are included?

Reply:
My understanding is the duplication could happen to prefetched dependencies by Cachi2. The application application source and the sources introduced by the original parent images like OS images should not duplicate.